### PR TITLE
Conversation fragments without text disable text copy.

### LINKED ISSUE #7472: Regression: No option in context menu to copy text of media message
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
It is no longer possible to copy the text of media messges. This behaviour is introduced by 4bdb2acd29f06ec0782f33e20b3c8290a002bef3.

### Steps to reproduce
- send a media message with text
- long-press this message

**Actual result:** No copy button available.
**Expected result:** A copy button should allow users to copy the text.

### Screenshots
bildschirmfoto von 2018-03-08 19-00-15

### Device info

**Device:** AVD Nexus 5X
**Android version:** 7.1.1
**Signal version:** 4.17.0

### Workaround
Use the forward button and select any conversation. Then copy the text from the text input box at the bottom of the conversation.

### PR DESCRIPTION
**Pull request by addumb**
### First time contributor checklist

- [x] I have read [how to contribute](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Nexus 5X, Android 8.1.0
 * Virtual device Pixel, Android 7.1.1
* Virtual device Nexus One, Android 7.1.1
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### Description

The multi-select redo from 4bdb2acd29f06ec0782f33e20b3c8290a002bef3 omitted the ability to copy text from media messages with text. It also removed the ability to select a plain text message plus a media message with text and copy the text from both. This fix just disables text copying whenever there's a message which has no text (event, media message without text).

Please pardon my n00bness. This is my first Android PR.

Fixes #7472

### CODE REVIEW

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 1]
```diff
@@ -230,19 +235,18 @@ private void setCorrectMenuVisibility(Menu menu) {
...
       MessageRecord messageRecord = messageRecords.iterator().next();
 
       menu.findItem(R.id.menu_context_resend).setVisible(messageRecord.isFailed());
-      menu.findItem(R.id.menu_context_save_attachment).setVisible(!actionMessage                     &&
-                                                                  messageRecord.isMms()              &&
-                                                                  !messageRecord.isMmsNotification() &&
-                                                                  ((MediaMmsMessageRecord)messageRecord).containsMediaSlide());
+      menu.findItem(R.id.menu_context_save_attachment).setVisible(!actionMessage &&
+                                                                  mediaMessage   &&
+                                                                  !textMessage);
```
Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: This means you cannot save media messages, which also include text, doesn't it?

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 2]
```diff
@@ -201,7 +201,8 @@ private void initializeListAdapter() {
   private void setCorrectMenuVisibility(Menu menu) {
     Set<MessageRecord> messageRecords = getListAdapter().getSelectedItems();
     boolean            actionMessage  = false;
-    boolean            mediaMessage  = false;
+    boolean            mediaMessage = false;
+    boolean            textMessage = true;
```
Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/ConversationFragment.java]: We don't really need `mediaMessage` AND `textMessage`. `mediaMessage` was introduced in #7410 to hide the copy button for media messages. But I already argued that I would prefer to hide the copy button depending on the absence of text in https://github.com/signalapp/Signal-Android/pull/7410#issuecomment-365702144. When we now decide to only hide the copy button, if there is no text, then we only need a variable to keep track of the existence of text.

In conclusion it should be sufficient to only have `textMessage` IF showing the copy button for media messages without actually copying the media is desired behaviour. You may look at #5857 as a template, if you want and @moxie0 requests you to make such a change.

File: src/org/thoughtcrime/securesms/ConversationFragment.java (hunk 1 above)
Comment (addumb) [reply src/org/thoughtcrime/securesms/ConversationFragment.java]: D'oh! Yeah. My bad.
